### DISCUSSION TITLE
Statement of Working | India | Patent

### DISCUSSION THREAD

**Discussion by OneHighCloud** [jjdejong/phpip#138] [Ideas]
After grant in India, a statement of working on Form-27 is required to be submitted. A financial year (FY) for India is 1st April - 31st March.

The Statement of working is required to be periodically submitted at Indian Patent Office (IPO) by 30th September of the 4th FY for previous 3 FYs, till the patent is active. The FY in which the patent is granted is exempted. For more clarity pls refer to **[Indian Patent Act, 1970, Section 146](https://ipindia.gov.in/writereaddata/Portal/ev/sections/ps146.html) and [Patents Rules, 2003, Rule 131](https://ipindia.gov.in/writereaddata/Portal/ev/rules/pr131.html)**. 

**This is a mandatory requirement in Indian Patent law**.

Rules engine is not able to create this task and deadline. So, I created a procedure for this purpose. If like, please feel free to use and incorporate in any future PHPIP update.

Starting:

**STEP 1:** Create an event "Statement of working" | Event Code "SAU" 
![Event](https://github.com/user-attachments/assets/ee51f4b4-2e3f-42d4-ba50-9277fc491a88)

**STEP 2:** Create a recurring rule: 
![Rule main](https://github.com/user-attachments/assets/995aec89-0544-4530-ac90-14352b5ae0f6)

![Rule condition](https://github.com/user-attachments/assets/aa3b5cbc-52da-44b8-b399-340d3791fb2f)

**Step 3:** CREATE Procedure (I ran MYSQL query)

DELIMITER $$
CREATE DEFINER=`phpip`@`%` PROCEDURE `insert_recurring_form27` (
    IN `P_trigger_id` INT, 
    IN `P_rule_id` INT, 
    IN `P_responsible` CHAR(16), 
    IN `P_user` CHAR(16)
)
proc: BEGIN
    DECLARE StartDate, DueDate, ExpiryDate DATE DEFAULT NULL;
    DECLARE GrantYear, FilingYear INT DEFAULT NULL;

    SELECT estart.event_date, matter.expire_date 
    INTO StartDate, ExpiryDate
    FROM matter
    JOIN event estart ON estart.matter_id = matter.id 
    AND estart.id = P_trigger_id;

    IF StartDate IS NULL THEN
        LEAVE proc;
    END IF;

    SET GrantYear = CASE 
        WHEN MONTH(StartDate) >= 4 
        THEN YEAR(StartDate) 
        ELSE YEAR(StartDate) - 1 
    END;

    SET FilingYear = GrantYear + 1;

    WHILE FilingYear <= YEAR(ExpiryDate) DO
        SET DueDate = STR_TO_DATE(CONCAT(FilingYear + 3, '-09-30'), '%Y-%m-%d');

        IF DueDate > ExpiryDate THEN
            LEAVE proc;
        END IF;

        INSERT INTO task (
            trigger_id, 
            code, 
            due_date, 
            detail, 
            rule_used, 
            assigned_to, 
            creator, 
            created_at, 
            updated_at
        ) 
        VALUES (
            P_trigger_id, 
            'SAU', 
            DueDate, 
            CONCAT('Form-27 filing for financial years ', FilingYear, '-', FilingYear + 3), 
            P_rule_id, 
            P_responsible, 
            P_user, 
            NOW(), 
            NOW()
        );

        SET FilingYear = FilingYear + 3;
    END WHILE;

END proc$$

DELIMITER ;

**STEP 4:** Make changes in "event_after_insert" trigger

From 

ELSEIF tr_task = 'REN' THEN
			-- Recurring renewal is the only possibilty here, normally
				CALL insert_recurring_renewals(NEW.id, tr_id, BaseDate, tr_responsible, NEW.creator);
			END IF;
		END LOOP create_tasks;

To

ELSEIF tr_task = 'REN' THEN			
				CALL insert_recurring_renewals(P_trigger_id, tr_id, BaseDate, tr_responsible, P_user);
            ELSEIF tr_task = 'SAU' THEN
            	CALL insert_recurring_form27(P_trigger_id, tr_id, tr_responsible, P_user);
			END IF;
	  END LOOP create_tasks;

**STEP 5:** Make changes in "recreate_tasks" procedure

From

ELSEIF tr_task = 'REN' THEN
			-- Recurring renewal is the only possibilty here, normally
				CALL insert_recurring_renewals(P_trigger_id, tr_id, BaseDate, tr_responsible, P_user);
			END IF;
	  END LOOP create_tasks;

To

ELSEIF tr_task = 'REN' THEN			
				CALL insert_recurring_renewals(NEW.id, tr_id, BaseDate, tr_responsible, NEW.creator);
            ELSEIF tr_task = 'SAU' THEN
            	CALL insert_recurring_form27(NEW.id, tr_id, tr_responsible, NEW.creator);
			END IF;
		END LOOP create_tasks;

### I am not a professional developer. So, please run tests. The above is working for me.

**Comment by jjdejong** [comment]
Thank you for this contribution. Indeed phpIP lacked the possibility of creating recurring tasks other than renewals. I'll try to find some time to test your contribution. The Indian situation being very specific, we didn't put resources into developing something.

As to "good practice" for events and tasks in phpIP, "Statement of Working" shouldn't be a task of its own (i.e. have a dedicated name). The task to create would rather be "Produce", with the detail "Statement of working" (which is how we implement this manually in our system). But I understand the statement of working is something similar to renewal fees, for which there is indeed a specific name.

By the way, why the code "SAU" for "Statement of Working"?

**Reply by OneHighCloud** [reply]
I am glad to be able to contribute something to this wonderful software. 

The due date for this "Statement of working" is fixed which is 30th September. Applicant/Licensee are required to submit the statement on or before this date. But, rules create deadlines based on trigger event and a predefined interval (days, months, years). Plus 3 years of recurrence was making it impossible to get correct deadlines for this task.

Because from original code, tasks != 'REN' and trigger event is 'GRT', this task is still listed under "Open Tasks" under "Granted" event header.

Original procedure "insert_recurring_renewals" is certainly a motivation for this procedure. 

This requirement has direct links to Compulsory License. Not only India but [many countries have similar compliances](https://www.obwb.com/newsletter/working-on-it-an-overview-of-patent-working-requirements-part-2) after grant, including some European members.

For the code, I started with "Statement After Utility grant". Therefore, the code "SAU". Later, description "Statement of Working" appeared more appropriate.

**Reply by papoteur-mga** [reply]
@OneHighCloud Thanks for the contribution.
I agree that this has to be dealt within the application. In practice, do you know how this is dealt by other software? Are they managed in the same workflow as renewals?

**Reply by iplaw67** [reply]
In Equinox (Questel), one would expect this kind of task to be a pre-set task based on the grant date, but as of yet (at least not in my instance), it isn't set up to generate them automatically. 
I've had to enter these manually at least for the first SOW date, for the Indian cases I'm managing in Equinox, with a 3-month and 1-month prior to the deadline reminder, based on the data provided by the Indian agent responsible for the case.
